packages/twenty-server/src/integrations/event-emitter/types/object-record-create.event.ts

```diff
@@ -1,6 +1,12 @@
 import { BaseObjectMetadata } from 'src/workspace/workspace-sync-metadata/standard-objects/base.object-metadata';
 
+export type CreatedObjectMetadata = {
+  nameSingular: string;
+  isCustom: boolean;
+};
+
 export class ObjectRecordCreateEvent<T extends BaseObjectMetadata> {
   workspaceId: string;
   createdRecord: T;
+  createdObjectMetadata: CreatedObjectMetadata;
```

Comment:
Why do you need this? Don't you have what you need inside createdRecord?

Reply:
No, the record contains only the data of the value, not the metadata

Reply:
Ah yes!